sponsorship_switzerland/models/contracts.py

```diff
@@ -413,3 +414,32 @@ def _reconcile_open_amount(self):
             delay = datetime.now() + relativedelta(seconds=15)
             if invoices:
                 invoices.with_delay(eta=delay).group_or_split_reconcile()
+
+    @api.multi
+    @job(default_channel='root.recurring_invoicer')
+    @related_action(action='related_action_contract')
+    def _clean_invoices(self, since_date=None, to_date=None, keep_lines=None):
+        today = datetime.now()
+        inv_lines = self.invoice_line_ids.filtered(lambda r: r.state == 'open'
+                                                   or (r.state == 'paid'
+                                                      and r.date > today))
```

Comment:
r.date is a string and today is a datetime. You should do `fields.Datetime.from_string(r.date) > today`